### PR TITLE
transducer functions and combinator

### PR DESCRIPTION
**Pull request by uriva**

### CODE REVIEW

File: gamla/transducer.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    return functional_generic.pipe(
+        funcs,
+        enumerate,
+        functional_generic.map(functional.star(_transform_nth)),
+        functional.star(functional_generic.compose),
+    )
+
+
+@currying.curry
+def mapping(f: Callable[[Any], Any], step: Reducer):
```
Comment (davidstrouk) [inline gamla/transducer.py:33]: imo a bit confusing to have `gamla.map` and `gamla.mapping`. 

Maybe call them `transmap`, `transfilter`, `transcat`,etc. ?

**Review by davidstrouk — APPROVED** [review]

**Review by davidstrouk — APPROVED** [review]

**Comment by davidstrouk** [comment]
@nitzo maybe worth you'd take a quick look too :)

File: gamla/transducer.py (hunk 1 above)
Comment (uriva) [reply gamla/transducer.py]: this is all pretty new, and eventually i think this is a better way to do `map` `filter` etc', so meanwhile I will just call it `map` and put it under the transducer namespace.

**Comment by uriva** [comment]
@nitzo ping

File: gamla/transducer.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    dict.items,
+    curried.map(functional.star(_transform_by_key(toolz.assoc))),
+    functional.star(functional_generic.compose),
+)
+
+
+juxt = functional_generic.compose_left(
+    functional.pack,
+    enumerate,
+    curried.map(functional.star(_transform_by_key(_replace_index))),
```
Comment (nitzo) [inline gamla/transducer.py:40]: gamla.map ?

File: gamla/transducer.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    )
+
+
+def _replace_index(the_tuple, index, value):
+    return (*the_tuple[:index], value, *the_tuple[index + 1 :])
+
+
+apply_spec = functional_generic.compose_left(
+    dict.items,
+    curried.map(functional.star(_transform_by_key(toolz.assoc))),
```
Comment (nitzo) [inline gamla/transducer.py:32]: gamla.map ?

**Review by nitzo — APPROVED** [review]

File: gamla/transducer.py (hunk 3 above)
Comment (uriva) [reply gamla/transducer.py:33]: i'm not sure we need the async pyrotechnics here just yet, so want to keep the dependencies in other parts of the lib minimal.

**Pull request merged**